internal/envoy/v3/cluster.go

```diff
@@ -339,6 +340,10 @@ func protocolOptions(explicitHTTPVersion HTTPVersionType, idleConnectionTimeout
 		options.CommonHttpProtocolOptions = &envoy_core_v3.HttpProtocolOptions{IdleTimeout: durationpb.New(idleConnectionTimeout.Duration())}
 	}
 
+	if maxRequestsPerConnection != nil {
+		options.CommonHttpProtocolOptions = &envoy_core_v3.HttpProtocolOptions{MaxRequestsPerConnection: wrapperspb.UInt32(*maxRequestsPerConnection)}
```

Comment:
looks like this will overwrite the above setting for idle timeout if both are set

Reply:
can add a test where both are set to make sure we cover this

Reply:
added